plugin/src/main/java/org/owasp/benchmarkutils/score/parsers/SnykReader.java

```diff
@@ -0,0 +1,112 @@
+                    put("PT", CweNumber.PATH_TRAVERSAL);
+                    put("HardcodedPassword", 0);
+                    put("WebCookieMissesCallToSetHttpOnly", CweNumber.INSECURE_COOKIE);
+                    put("ServerInformationExposure", 0);
+                    put("UserControlledFormatString", CweNumber.EXTERNALLY_CONTROLLED_STRING);
+                    put("SpringCSRF", CweNumber.CSRF);
+                    put("TrustBoundaryViolation", CweNumber.TRUST_BOUNDARY_VIOLATION);
+                    put("CommandInjection", CweNumber.COMMAND_INJECTION);
+                    put("DOMXSS", CweNumber.XSS);
+                    put("XSS", CweNumber.XSS);
+                    put("InsecureCipherNoIntegrity", CweNumber.WEAK_CRYPTO_ALGO);
+                    put("InsecureDefaultAesCipher", CweNumber.WEAK_CRYPTO_ALGO);
+                    put("HttpResponseSplitting", CweNumber.HTTP_RESPONSE_SPLITTING);
+                    put("InsecureSecret", CweNumber.WEAK_RANDOM);
+                    put("LdapInjection", CweNumber.LDAP_INJECTION);
+                    put("InsecureCipher", CweNumber.WEAK_CRYPTO_ALGO);
+                    put("InsecureHash", CweNumber.WEAK_HASH_ALGO);
+                }
+            };
+
+    @Override
+    public boolean canRead(ResultFile resultFile) {
+        return resultFile.isJson() && isSnyk(resultFile);
+    }
+
+    @Override
+    public TestSuiteResults parse(ResultFile resultFile) throws Exception {
+        TestSuiteResults tr = new TestSuiteResults("Snyk", true, TestSuiteResults.ToolType.SAST);
+
+        JSONArray results =
+                resultFile.json().getJSONArray("runs").getJSONObject(0).getJSONArray("results");
+
+        for (int result = 0; result < results.length(); result++) {
+            TestCaseResult tcr = parseSnykFindings(results.getJSONObject(result));
+            if (tcr != null) {
+                tr.put(tcr);
+            }
+        }
+        return tr;
+    }
+
+    private TestCaseResult parseSnykFindings(JSONObject result) {
+        try {
+            String className =
+                    result.getJSONArray("locations")
+                            .getJSONObject(0)
+                            .getJSONObject("physicalLocation")
+                            .getJSONObject("artifactLocation")
+                            .getString("uri");
+            className = (className.substring(className.lastIndexOf('/') + 1)).split("\\.")[0];
+            if (className.startsWith(BenchmarkScore.TESTCASENAME)) {
+
+                TestCaseResult tcr = new TestCaseResult();
+
+                String ruleId = result.getString("ruleId");
+                ruleId = (ruleId.substring(ruleId.lastIndexOf('/') + 1)).split("\\.")[0];
+
+                int cwe = snykCweMap.getOrDefault(ruleId, INVALID_RULE_ID);
+
+                if (cwe == INVALID_RULE_ID) {
+                    System.out.println("CWE # not parseable from: " + ruleId);
+                    return tcr;
```

Comment:
If the CWE number couldn't be read, I guess you should just return null. Right @davewichers ?

Reply:
I believe that's right. If null is returned the rest case result is not added to the results set, which is what you want. You should test it to make sure it doesn't throw an NPE, but I think that's right.

Reply:
What if a new ruleId is added to snyk code?
If a new ruleId is added, then the tool will always throw NullPointerException. So, i felt that adding an empty testCaseResult is a better approach. 

This is my opinion. What do you folks think? @darkspirit510 @davewichers

Reply:
Did you try this? Other readers check for null return and skip adding it to the collection 🤔